### PR TITLE
#2742 number formatters

### PR DESCRIPTION
**Pull request by AlexeyTuboltsev**
this is my suggestion how to refactor components used to render numbers and money. It doesn't affect the old code, just introduces new components and a couple of tests to show the usage.
The future replacement for Money is called MoneyNew for now :)

I suggest to leave FormatNumber, formatShortNumber and Money (MoneyNew) components and additionally introduce FormatNumberRange and MoneyRange. Money, MoneyRange and FormatNumberRange internally render FormatNumber OR formatShortNumber but with preset rendering options like rounding, separators, fallbacks etc. 

As there's a finite set of valid possibilities to render money (see table below), I suggest to create an Enum that we pass to Money/MoneyRange, 
The values that we render can be in an ulps or float format. This is another enum.
The output format is a set too (like abbreviated (like 25k Eur) or full (25 000.00 Euro)). This is the third enum

Together these three enums (inputFormat, moneyFormat, outputFormat) must cover all possibilities to render money. All settings (rounding, currency sign, isPrice etc) are set internally inside the Money component basing on this enum, leaving much less possibilities to introduce an error or confuse developers.

If we need to display money in a custom one-off way, or to simply render a number, please use a FormatNumber/ FormatNumberRange component that then would need all these settings explicitly provided. 

If we find out that there are possibilities missing, we'd need to expand the enums, not add custom parameters

🤘 

here are the number/money guidelines:
Screen Shot 2019-04-16 at 17 29 26

Fun fact: here is a list of functions and components we use now to render numbers/money (some of them sometimes call the others internally):
formatFlexiPrecision
formatThousands
formatVaryingDecimals
formatEthTsd
formatEurTsd
formatThousands
formatEur
formatMoney
formatToFixed
getFormattedMoney
toHumanReadableForm
Money

### CODE REVIEW

**Comment by AlexeyTuboltsev** [comment]
hm. the moneyFormat and outputFormat could be actually merged to one enum, to simplify things even further

File: app/components/shared/Money.unsafe.tsx [hunk 1]
```diff
@@ -101,7 +101,10 @@ function getFormatDecimals(format: EMoneyFormat): number {
       throw new Error("Unsupported money format");
   }
 }
-
+/*
+* @deprecated
```
Comment (Moejoe90) [inline app/components/shared/Money.unsafe.tsx]: :+1:

File: app/components/shared/formatters/utils.ts [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+  } else {
+    return value;
+  }
+};
+
+//TODO sorry for any, I'll figure out how to type it later. Bignumber doesn't expose RoundingMode there's no easy solution for it
+export function getBigNumberRoundingMode(
+  roundingMode: ERoundingMode,
+  outputFormat: EHumanReadableFormat = EHumanReadableFormat.FULL,
+): any {
```
[the remark this answers is not in the document]
Comment (AlexeyTuboltsev) [reply app/components/shared/formatters/utils.ts]: left as is for now

File: app/components/shared/formatters/utils.spec.ts [hunk 3]
```diff
@@ -1,12 +1,160 @@
...
   isValidNumberOrEmpty,
   parseInputToNumber,
+  removeZeroDecimals,
+  selectDecimalPlaces,
   stripNumberFormatting,
-} from "./Money.utils";
+  toFixedPrecision,
+} from "./utils";
+
+describe("formatNumber", () => {
```
[the remark this answers is not in the document]
Comment (AlexeyTuboltsev) [reply app/components/shared/formatters/utils.spec.ts]: done

File: app/components/shared/MaskedMoneyInput.tsx [hunk 4]
```diff
@@ -1,16 +1,19 @@
-import { BigNumber } from "bignumber.js";
```
[the remark this answers is not in the document]
Comment (AlexeyTuboltsev) [reply app/components/shared/MaskedMoneyInput.tsx]: let's postpone it, i'll add it after refactoring redeem component

File: app/components/shared/MaskedMoneyInput.tsx [hunk 5]
```diff
@@ -26,59 +29,55 @@ interface IProps {
...
 export class MaskedMoneyInput extends React.Component<IProps> {
-  moneyToPrimaryBase = (value: string | BigNumber | number, currencyDecimals: number) => {
-    const money = value instanceof BigNumber ? value : new BigNumber(value.toString());
-    return money.div(new BigNumber(10).pow(currencyDecimals)).toString();
-  };
+  decimals = selectDecimalPlaces(this.props.currency);
 
-  formatMoney = (value: string, decimalPlaces: number): string => {
+  formatValue = (value: string): string => {
     if (isValidNumberOrEmpty(value)) {
```
[the remark this answers is not in the document]
Comment (AlexeyTuboltsev) [reply app/components/shared/MaskedMoneyInput.tsx]: you're right
split this function in two, added unit test specific for both

**Pull request merged**